### PR TITLE
[WIP] Email subscriptions, better RSS subscription UI

### PR DESCRIPTION
**Pull request by jimrandomh**

### CODE REVIEW

**Comment by jimrandomh** [comment]
This PR incorporates and continues Adrian's PR https://github.com/LessWrong2/Lesswrong2/pull/824

**Comment by jimrandomh** [comment]
Requested change: if you're not logged in, the "you must be logged in to subscribe" should instead be a message with login/register links. We can pattern this off the "you must be logged in to comment" link, but that was broken by the Bootstrap removal.

**Comment by jimrandomh** [comment]
* The subscribe links should have the `cursor:pointer` style.
 * The `Subscribe` text next to the icons should also be a link
 * Disable options other than Curated in the Email subscribe tab, unless/until other feeds are email-able

**Comment by jimrandomh** [comment]
* If Timber logging is enabled, there's a crash with a stacktrace that passes through Timber.
 * In `maybeSendVerificationEmail` there's a crash if the old `whenConfirmationEmailSent` is null

**Comment by jimrandomh** [comment]
* If you go from verifying your email to the Edit Account page without a refresh, Apollo has an out of date view of the user and thinks it's not confirmed.

**Comment by jimrandomh** [comment]
* If you're already email-subscribed to a feed, the button label should reflect that

**Comment by jimrandomh** [comment]
* The login link on the subscribe dialog doesn't look like a link

File: packages/lesswrong/components/common/SubscribeDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+  }
+
+  subscribeByEmail() {
+    let mutation = { emailSubscribedToCurated: true }
+
+    if (!Users.emailAddressIsVerified(this.props.currentUser)) {
+      // Updating whenConfirmationEmailSent sets off a trigger
+      // which causes the email to actually be sent.
+      //
+      // We combine these into one editMutation call to work
```
Comment (Discordius) [inline packages/lesswrong/components/common/SubscribeDialog.jsx]: Change this comment to reflect change to sync callback

File: packages/lesswrong/lib/modules/callbacks.js [hunk 2]
```diff
@@ -63,6 +64,24 @@ const createNotifications = (userIds, notificationType, documentType, documentId
   });
 }
 
+const sendEmailPostNotifications = async (users, notificationType, postId) => {
```
Comment (Discordius) [inline packages/lesswrong/lib/modules/callbacks.js]: Rename to not use "notifications" and also get rid of unused `notificationType`

**Comment by Discordius** [comment]
* Add `8 px` top margin for the Subscribe widget 
* Fix the submit button the Edit User page

**Comment by jimrandomh** [comment]
* There should be another check for email-address confirmation at send time

**Comment by jimrandomh** [comment]
* Change verification email subject line
 * Change "check your inbox for a verification link" message to something clearer

**Comment by jimrandomh** [comment]
* Email template should have some boilerplate at the bottom about your subscription

**Pull request merged**